File: src/test/java/com/meti/DeclarationFeatureTest.scala
package com.meti

import com.meti.MagmaAssertions.assertCompile
import org.junit.jupiter.api.Test

class DeclarationFeatureTest {
	@Test
	def with_both(): Unit =
		assertCompile("int x=10", "const x : I16 = 10;")
}

File: src/test/java/com/meti/compile/feature/declare/DeclarationFeatureTest.java
package com.meti.compile.feature.declare;

import org.junit.jupiter.api.Test;

import static com.meti.MagmaAssertions.assertCompile;
import static com.meti.MagmaAssertions.assertCompileThrows;

class DeclarationFeatureTest {
	@Test
	void already_exists() {
		assertCompileThrows("const x=420;const x=10;");
	}

	@Test
	void implicit() {
		assertCompile("const x=420", "signed int x=420;");
	}

	@Test
	void positive() {
		assertCompile("const x : I16 = 420", "signed int x=420;");
	}
}
